pynicotine/gtkgui/chatrooms.py

```diff
@@ -658,14 +658,14 @@ def read_room_logs_finished(self):
 
         self.activity_view.auto_scroll = self.chat_view.auto_scroll = True
 
-    def read_room_logs(self):
+    def prepend_old_messages(self):
 
-        self.chat_view.append_log_lines(
-            folder_path=log.room_folder_path,
-            basename=self.room,
-            num_lines=config.sections["logging"]["readroomlines"],
-            timestamp_format=config.sections["logging"]["rooms_timestamp"]
-        )
+        log_lines = self._read_old_messages()
+
+        self.chat_view.append_log_lines(log_lines, login_username=config.sections["server"]["login"])
+
+    def _read_old_messages(self):
```

Comment:
Would probably make more sense to get rid of this function for now and just call log.read_log() directly?

Reply:
Yes, it's not required until textview is made capable of prepending lines as in PR #2792. In there, I found that a seg fault may occur when rapidly infinite scrolling (such as when holding down the PageUp key) unless GLib.idle_add() was used, which doesn't accept multiple arguments hence why this separate function call was implemented.